### PR TITLE
Implemented date_part in where filter

### PR DESCRIPTION
**Pull request by DevonFulcher**
### Description

This PR allows users to specify `date_part` in the where filter. This is useful for the Jinja syntax as well as integrations such as Tableau.

### CODE REVIEW

**Review by tlento — COMMENTED** [review]
This looks like two PRs in one. Is there something wonky in the underlying branch, or should this be split up?

**Comment by DevonFulcher** [comment]
> This looks like two PRs in one. Is there something wonky in the underlying branch, or should this be split up?

I fixed it now. I'm sorry about that.

File: metricflow/specs/where_filter_dimension.py
Comment (tlento) [inline metricflow/specs/where_filter_dimension.py:48]: Ah, so this returns a TimeDimension with DEFAULT_TIME_GRANULARITY if neither grain nor date_part are set. Interesting. I think that makes the most sense with where we're heading with time dimension expressions.

File: metricflow/test/model/test_where_filter_spec.py
Comment (tlento) [inline metricflow/test/model/test_where_filter_spec.py:131]: I hope these aren't case-sensitive. I may add some additional tests on after we merge.

**Review by tlento — COMMENTED** [review]
Looks great, thanks! I'll hold this open until after 0.203.0 is out (unless that drags past Monday, in which case maybe we just merge this).

**Review by tlento — APPROVED** [review]

**Pull request merged**
